controllers/complianceeventsapi/server.go

```diff
@@ -528,17 +558,82 @@ func getSingleComplianceEvent(db *sql.DB, w http.ResponseWriter, r *http.Request
 	}
 }
 
+func getClusterNameFromID(ctx context.Context, db *sql.DB, clusterID string) (name string) {
+	err := db.QueryRowContext(ctx,
+		`SELECT name FROM clusters WHERE cluster_id = $1`, clusterID,
+	).Scan(&name)
+	if err != nil {
+		return ""
+	}
+
+	return name
+}
+
 // getWhereClause will convert the input queryOptions to a WHERE statement and return the filter values for a prepared
 // statement.
-func getWhereClause(options *queryOptions) (string, []any) {
+func getWhereClause(ctx context.Context, options *queryOptions,
+	db *sql.DB, config *rest.Config, token string, client client.Client,
+) (string, []any, error) {
 	filterSQL := []string{}
 	filterValues := []any{}
 
+	allManagedClusterRules, err := getAllManagedClusterRules(ctx, config, token, client)
+	if err != nil {
+		return "", nil, err
+	}
+
+	var filterClusterNames []string
+
+	// "True" means the user added "cluster_id" or "cluster.name" as url filters
+	hasClusterFilter := false
+
 	for sqlColumn, values := range options.Filters {
 		if len(values) == 0 {
 			continue
 		}
 
+		// "cluster" change to "clusters" at queryOptionsToSQL
+		if sqlColumn == "clusters.cluster_id" {
```

Comment:
I'm thinking that it'd be cleaner if `parseQueryArgs` did the auth check and set the `clusters.cluster_id` or `clusters.name` filter in `options.Filters` with what the user has access to.

Then `getWhereClause` would not need any changes.

Reply:
@mprahl  When the user has only partial auth for clusters,  should we send a message for that?